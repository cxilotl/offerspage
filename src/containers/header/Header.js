import React from 'react';
import cssStyles from './Header.module.scss';
import logo from "../../logo-horizontal.svg";
import AccountButton from '../../components/accountButton/AccountButton';
import LocationWidget from '../../components/locationWidget/LocationWidget';

const Header = () => {
  const user = 'Jane Smith';
  const handleAccountBtnClicked = () => {
    console.log('Open Account'); // eslint-disable-line no-console
  };
  return(
    <header className={ cssStyles.header }>
      <section className={ cssStyles.sectionMain }>
        <a href="/" className={ cssStyles.linkLogo }>
          <h1>
            <img src={ logo } className={ cssStyles.logo } alt="Deliveroo logo" />
          </h1>
        </a>
        <AccountButton userName={ user } onCLick={ handleAccountBtnClicked } />
      </section>

      <section className={ cssStyles.sectionLocation }>
        <LocationWidget />
      </section>
    </header>
  );
};

export default Header;