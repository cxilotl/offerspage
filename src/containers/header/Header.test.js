import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';

describe('Header', () => {

  it('Snapshot', () => {
    const component = renderer.create(
      <Header />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});