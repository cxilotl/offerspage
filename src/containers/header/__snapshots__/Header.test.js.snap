// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot 1`] = `
<header
  className="header"
>
  <section
    className="sectionMain"
  >
    <a
      className="linkLogo"
      href="/"
    >
      <h1>
        <img
          alt="Deliveroo logo"
          className="logo"
          src="logo-horizontal.svg"
        />
      </h1>
    </a>
    <button
      className="layout"
      data-testid="account-button"
      id="account-btn"
      name="account-btn"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="User icon"
        className="icon"
        src="user-icon.svg"
      />
      <span
        className="label"
      >
        Jane Smith
      </span>
    </button>
  </section>
  <section
    className="sectionLocation"
  >
    <div
      className="mainInfo"
    >
      <span
        className="label"
      >
        Location
      </span>
      <span
        className="selectedLocation"
      >
        Fritzovia
      </span>
    </div>
    <button
      className="locationSelectionBtn"
      data-testid="location-button"
      id="locationUpdate"
      name="locationUpdate"
      onClick={[Function]}
      type="button"
    >
      Change Location
    </button>
  </section>
</header>
`;
